File: 프로그래머스/0/120896. 한 번만 등장한 문자/한 번만 등장한 문자.py
def solution(s):
    answer = []
    box = []                    
    for i in s:
        if i in answer:
            answer.remove(i)
            box.append(i)

        elif i not in answer and i in box:
            box.append(i)

        else:
            answer.append(i)
            box.append(i)

    return "".join(sorted(answer))

File: 프로그래머스/0/120896. 한 번만 등장한 문자/한 번만 등장한 문자.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Comparator;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;
import java.util.stream.Collector;
import java.util.stream.Collectors;
import java.util.stream.IntStream;
import java.util.stream.Stream;
import java.util.Scanner;
class Solution {
    public String solution(String s) {
    List <String> answer = new ArrayList<>(); 
    Map <String,Integer> counter = new HashMap<>();
    for (char i : s.toCharArray()){
        String key = String.valueOf(i);
        counter.put(key, counter.getOrDefault(key, 0) + 1);
    }

    for (String i : counter.keySet()){
        if(counter.get(i) == 1){
            answer.add(i);
        }
    }
    return answer.isEmpty() ? "" : answer.stream().sorted().collect(Collectors.joining(""));
}
}